pkg/ir/spec.go

```diff
@@ -161,164 +214,96 @@ func (n *NifDetails) Address() *netset.IPBlock {
-	for i, vpeEndPoint := range VPEDetails.VPEReservedIPs {
-		ips[i] = s.VPEReservedIPs[vpeEndPoint].IP
-	}
-	return Resource{name, ips, ResourceTypeVPE}, nil
+func (i *InstanceDetails) endpointType() ResourceType {
+	return ResourceTypeNIF
 }
 
-func (s *Definitions) lookupSubnetSegment(name string) (Resource, error) {
-	if subnetSegmentDetails, ok := s.SubnetSegments[name]; ok {
-		cidrs := make([]*netset.IPBlock, len(subnetSegmentDetails.Subnets))
-		for i, subnetName := range subnetSegmentDetails.Subnets {
-			subnet, err := s.Lookup(ResourceTypeSubnet, subnetName)
-			if err != nil {
-				return Resource{}, fmt.Errorf("%w while looking up %v %v for subnet %v", err, ResourceTypeSubnet, subnetName, name)
-			}
-			// each subnet has only one CIDR block.
-			cidrs[i] = subnet.IPAddrs[0]
-		}
-		return Resource{name, cidrs, ResourceTypeSubnet}, nil
-	}
-	return Resource{}, containerNotFoundError(name, ResourceTypeSegment)
+func (v *VPEDetails) endpointNames() []ID {
+	return v.VPEReservedIPs
 }
 
-func (s *Definitions) lookupCidrSegment(name string) (Resource, error) {
-	cidrSegmentDetails, ok := s.CidrSegments[name]
-	if !ok {
-		return Resource{}, containerNotFoundError(name, ResourceTypeSegment)
+func (v *VPEDetails) endpointMap(s *Definitions) map[ID]INWResource {
+	res := make(map[ID]INWResource, len(s.VPEReservedIPs))
+	for k, v := range s.VPEReservedIPs {
+		res[k] = v
 	}
-	return Resource{name, cidrSegmentDetails.Cidrs.SplitToCidrs(), ResourceTypeCidr}, nil
+	return res
 }
 
-func (s *Definitions) Lookup(t ResourceType, name string) (Resource, error) {
-	err := fmt.Errorf("invalid type %v (resource %v)", t, name)
-	switch t {
-	case ResourceTypeExternal:
-		return lookupSingle(s.Externals, name, t)
-	case ResourceTypeSubnet:
-		return lookupSingle(s.Subnets, name, t)
-	case ResourceTypeCidr:
-		return lookupSingle(s.Subnets, name, t)
-	case ResourceTypeNIF:
-		return lookupSingle(s.NIFs, name, t)
-	case ResourceTypeVPE:
-		return s.lookupVPE(name)
-	case ResourceTypeInstance:
-		return s.lookupInstance(name)
-	case ResourceTypeSegment:
-		if _, ok := s.SubnetSegments[name]; ok { // subnet segment
-			return s.lookupSubnetSegment(name)
-		} else if _, ok := s.CidrSegments[name]; ok { // cidr segment
-			return s.lookupCidrSegment(name)
-		} else {
-			return Resource{}, err
-		}
-	default:
-		return Resource{}, err
-	}
+func (v *VPEDetails) endpointType() ResourceType {
+	return ResourceTypeVPE
 }
 
-func inverseLookup[T NWResource](m map[ID]T, address *netset.IPBlock) (result string, ok bool) {
-	for name, details := range m {
-		if details.Address().Equal(address) {
-			return name, true
-		}
+func lookupSingle[T NWResource](m map[ID]T, name string, t ResourceType) (*LocalRemotePair, error) {
+	if details, ok := m[name]; ok {
+		return &LocalRemotePair{
+			Name:        &name,
+			LocalCidrs:  []*NamedAddrs{{Name: &name, IPAddrs: details.Address()}},
+			RemoteCidrs: []*NamedAddrs{{Name: &name, IPAddrs: details.Address()}},
```

Comment:
Why do we set `RemoteCidrs` here? It looks a bit strange.
Please add a comment.

Reply:
why does it look strange?

Reply:
Because `RemoteCidrs` is set to exactly the same value as `LocalCidrs`. One might expect local and remote to be different.